app/volume/route.js

```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  queryParams: {
+    type: {
+      refreshModel: true
+    },
+  },
+  model: function(params) {
+    return this.get('store').getById(params.type, params.volume_id);
+  },
+  setupController(controller, model) {
+    this._super(controller, model);
```

Comment:
Any particular reason to setupController vs returning multiple properties in the model?

As-is, the stack and host aren't being cleared for the next time you use this controller to view a different volume (which may or may not matter).